plugin/evm/limitorders/events.go

```diff
@@ -0,0 +1,122 @@
+package limitorders
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+func (lotp *limitOrderTxProcessor) HandleOrderBookEvent(event *types.Log) {
+	args := map[string]interface{}{}
+	switch event.Topics[0] {
+	case lotp.orderBookABI.Events["OrderPlaced"].ID:
+		err := lotp.orderBookABI.UnpackIntoMap(args, "OrderPlaced", event.Data)
+		if err != nil {
+			log.Error("error in orderBookAbi.UnpackIntoMap", "method", "OrderPlaced", "err", err)
+		}
+		log.Info("HandleOrderBookEvent", "orderplaced args", args)
+		order := getOrderFromRawOrder(args["order"])
+
+		lotp.memoryDb.Add(&LimitOrder{
+			Market:            Market(order.AmmIndex.Int64()),
+			PositionType:      getPositionTypeBasedOnBaseAssetQuantity(order.BaseAssetQuantity),
+			UserAddress:       getAddressFromTopicHash(event.Topics[1]).String(),
+			BaseAssetQuantity: order.BaseAssetQuantity,
+			Price:             order.Price,
+			Status:            Unfulfilled,
+			RawOrder:          args["order"],
+			// RawSignature:      args["signature"],
+			Signature:         args["signature"].([]byte),
+			BlockNumber:       big.NewInt( int64(event.BlockNumber)),
+		})
+	case lotp.orderBookABI.Events["OrdersMatched"].ID:
+		log.Info("OrdersMatched event")
+		err := lotp.orderBookABI.UnpackIntoMap(args, "OrdersMatched", event.Data)
+		if err != nil {
+			log.Error("error in orderBookAbi.UnpackIntoMap", "method", "OrdersMatched", "err", err)
+		}
+		log.Info("HandleOrderBookEvent", "OrdersMatched args", args)
+		signature1 := args["signature1"].([]byte)
+		signature2 := args["signature2"].([]byte)
+		fillAmount := args["fillAmount"].(*big.Int)
+		lotp.memoryDb.UpdateFilledBaseAssetQuantity(fillAmount, signature1)
```

Comment:
curious why we index with signature and not with orderHash

Reply:
back when we were parsing contract transactions instead of events, the functions were being called using signature and order object. To calculate hash, we would have had to call getOrderHash in the contract. So we started using signature instead. At this point we can use hash too if it's emitted in OrderPlaced and OrdersMatched